samples/boards/nrf52/mesh/onoff_level_lighting_vnd_app/src/mesh/state_binding.c

```diff
@@ -0,0 +1,206 @@
+/* Bluetooth: Mesh Generic OnOff, Generic Level, Lighting & Vendor Models
+ *
+ * Copyright (c) 2018 Vikrant More
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <board.h>
+#include <gpio.h>
+
+#include "common.h"
+#include "ble_mesh.h"
+#include "device_composition.h"
+
+#define MINDIFF 2.25e-308
+
+static float sqrt(float square)
+{
+	float root, last, diff;
+
+	root = square / 3.0;
+	diff = 1;
+
+	if (square <= 0) {
+		return 0;
+	}
+
+	do {
+		last = root;
+		root = (root + square / root) / 2.0;
+		diff = root - last;
+	} while (diff > MINDIFF || diff < -MINDIFF);
+
+	return root;
+}
+
+static s32_t ceiling(float num)
+{
+	s32_t inum;
+
+	inum = (s32_t) num;
+	if (num == (float) inum) {
+		return inum;
+	}
+
+	return inum + 1;
+}
+
+static bool constrain_light_actual_state(u16_t var)
+{
+	bool is_value_within_range;
+
+	is_value_within_range = false;
+
+	if (var > 0 &&
+	    var < light_lightness_srv_user_data.light_range_min) {
+		var = light_lightness_srv_user_data.light_range_min;
+	} else if (var > light_lightness_srv_user_data.light_range_max) {
+		var = light_lightness_srv_user_data.light_range_max;
+	} else {
+		is_value_within_range = true;
+	}
+
+	light_lightness_srv_user_data.actual = var;
+
+	return is_value_within_range;
+}
+
+static void update_gen_onoff_state(void)
+{
+	if (light_lightness_srv_user_data.actual == 0) {
+		gen_onoff_srv_root_user_data.onoff = STATE_OFF;
+	} else {
+		gen_onoff_srv_root_user_data.onoff = STATE_ON;
+	}
+}
+
+void state_binding(u8_t lightness, u8_t temperature)
+{
+	bool bin;
+	u16_t tmp16;
+	float tmp;
```

Comment:
Please stop using all these tmp variables and name your variables descriptively. It's really hard to follow the code otherwise. Should this have been `temp` instead of `tmp`? Same thing with tmp16.

Reply:
Why temp ?? ...it has used at multiple location.
It is there from beginning onwards. So why this sudden requirement pops up ?

Reply:
> It is there from beginning onwards. So why this sudden requirement pops up ?

The answer is really quite simple: I'm not a machine.

Your app is not small, and it just keeps growing. I have limited time to go through it and will for sure keep missing out on things in the future as well.

Reply:
I'm completely agree with you. But currently if possible ignore it. I will do such make-up related things via separate PR once we pass test PTS. 

And thanks for considering this App is not a small app.... :)